### ISSUE TITLE
Bug: Incorrect model type and running mode for Curtain 3

### ISSUE THREAD

**Issue by hqy2000** [OpenWonderLabs/homebridge-switchbot#893]
### Describe the Bug

1. Curtain 3 uses `model: '{'` instead of `model: 'c'` for `startScan` and `discover`, according to https://github.com/OpenWonderLabs/node-switchbot/blob/031d047d26656528eef22c6f582ba04df8a56e3b/lib/switchbot.js#L335. This causes the plugin to timeout during scanning and discovering, producing error messages related to not being able to establish a BLE connection.

2. Curtain 3 uses number `1` for QuietDrift mode, `255` for normal mode in `runToPos`: https://github.com/OpenWonderLabs/node-switchbot/blob/031d047d26656528eef22c6f582ba04df8a56e3b/lib/switchbot-device-wocurtain.js#L79. It seems that `curtain.ts` uses an object for that, causing the method to throw exceptions.

**I was able to get both to work by manually changing all `c` into `{` and set the 2nd parameter to `255` for `runToPos` in `curtain.ts`.**

### To Reproduce

1. Install a Curtain 3.
2. Configure the plugin with a Curtain 3 bot only using BLE connections.
4. Start the plugin, you'll see error messages `... wasn't able to establish BLE Connection, node-switchbot: [object Object]` due to incorrect model type.

### Expected Behavior

The plugin should correctly communicate with the bot. 

### Relevant Log Output

```shell
... wasn't able to establish BLE Connection, node-switchbot: [object Object]
... The type of running mode is incorrect: object
```

### Config for homebridge-switchbot

```shell
{
    "name": "SwitchBot",
    "credentials": {
        "notice": "Keep your Token & Secret a secret!"
    },
    "options": {
        "devices": [
            {
                "deviceId": "xxxxxxxxxxxx",
                "configDeviceName": "Curtain B6",
                "configDeviceType": "Curtain",
                "connectionType": "BLE"
            }
        ]
    },
    "platform": "SwitchBot"
}
```

### Device and Model

Curtain 3

### Node.js Version

v20.10.0

### NPM Version

v10.2.3

### Homebridge Version

v1.7.0

### Homebridge Switchbot Plugin Version

v2.14.0

### Operating System

Rasbian

**Comment by donavanbecker** [comment]
Then you should open. A PR on node-SwitchBot with the fix

**Comment by hqy2000** [comment]
I believe `model` comes directly from `serviceData`? So does that mean it's hardcoded in the firmware? 
https://github.com/OpenWonderLabs/node-switchbot/blob/031d047d26656528eef22c6f582ba04df8a56e3b/lib/switchbot-advertising.js#L84

**Comment by donavanbecker** [comment]
not sure, but the data that is received from the bytes could be manipulated and changed with node-switchbot

https://github.com/OpenWonderLabs/node-switchbot/blob/031d047d26656528eef22c6f582ba04df8a56e3b/lib/switchbot-advertising.js#L97

**Comment by hqy2000** [comment]
Unfortunately, I can’t fix this due to time constraints. If anyone else can take it on, that would be great. Thanks!

**Comment by donavanbecker** [comment]
Fixed with v3.1.0

I have added Curtain 3 to the config options, this will now allow you to send `{` instead of `c`. Once you install the beta, update your DeviceType to `Curtain3`.
